src/template_partials/templatetags/partials.py

```diff
@@ -53,17 +53,19 @@ def render(self, context):
 
 
 class RenderPartialNode(template.Node):
-    def __init__(self, partial_name, origin):
+    def __init__(self, partial_name):
         self.partial_name = partial_name
-        self.origin = origin
 
     def render(self, context):
-        """Render the partial content from the context"""
-        # Use the origin to get the partial content, because it's per Template,
-        # and available to the Parser.
-        # TODO: raise a better error here.
-        nodelist = self.origin.partial_contents[self.partial_name]
-        return nodelist.render(context)
+        template_obj = getattr(context, "template", None)
```

Comment:
OK, so this is a change in strategy. (Might well be for the good!)

Under what circumstances are you expecting `context.template` to be missing? 🤔